### DISCUSSION TITLE
Mesh adaptivity adds a huge nonlinear residual jump

### DISCUSSION THREAD

**Discussion by xueyang94** [idaholab/moose#17551] [Q&A Modules: Phase field] [answered]
Hi everyone,

I am running a KKS model with the Newton solver. Normally the nonlinear residual is around 1e1. After adding a mesh adaptivity in the Executioner block however, the nonlinear residual residual at the first iteration jumps to 1e8. The adaptivity looks like below. Could anyone help to explain why this happens? Thanks in advance. 

[./Adaptivity]
    initial_adaptivity = 1 
    interval = 1
    refine_fraction = 0.9
    coarsen_fraction = 0.1
    max_h_level = 4
 [../]

**Comment by simopier** [comment, marked as answer]
Hi Xueyang, 

You seem to have defined a maximum of 4 levels of mesh adaptivity `max_h_level = 4`, but allowed the initial step to include only one level `initial_adaptivity = 1`. Maybe your mesh is very coarse during your initial step, which leads to large residuals.

To test your code and make sure the transition to mesh adaptivity goes smoothly, I suggest doing it step by step from a simple input file that runs well without mesh adaptivity.

Once you have that, you could introduce one level of mesh adaptivity. In your mesh block, introduce a mesh that is twice as coarse (`nx` and `ny` defined as half of what you were using without mesh adaptivity for a 2D mesh) and add `uniform_refine = 1`. This line will uniformly refine your mesh once, thus creating the same mesh that you had in the input file without mesh adaptivity. Then, to introduce mesh adaptivity, and define your mesh adaptivity block as 
```
[./Adaptivity]
initial_adaptivity = 1
interval = 1
refine_fraction = 0.9
coarsen_fraction = 0.1
max_h_level = 1
[../]
```
Notice as I defined `max_h_level = 1` instead of 4, since we want to introduce only one level of mesh adaptivity. If you check your mesh, some parts might be twice coarse as the uniform mesh you had without mesh adaptivity.

Let me know if this solves your issue. If it does, you can progressively increase the number of mesh adaptivity levels by increasing `initial_adaptivity` and `max_h_level`.
